### DISCUSSION TITLE
Deploying channels with azure cache for redis as host - redis.exceptions.ConnectionError: Error while reading from socket: (104, 'Connection reset by peer')

### DISCUSSION THREAD

**Discussion by vishwakarma0** [django/channels#1807] [Help!] [answered]
redis.exceptions.ConnectionError: Error while reading from socket: (104, 'Connection reset by peer')

-  OS used is Ubuntu
- packages used
    channels-redis==3.3.1
    hiredis==2.0.0
    redis==2.10.6
    channels==3.0.4
    channels-redis==3.3.1
When running with the localhost it runs smoothly, but when I try to use azure cache for redis which is password protected, it throws the below error
`redis.exceptions.ConnectionError: Error while reading from socket: (104, 'Connection reset by peer')
`
channel configuration in settings.py
`CHANNEL_LAYERS = {
    'default': {
        'BACKEND': 'channels_redis.core.RedisChannelLayer',
        'CONFIG': {
            "hosts": [("redis://:mypassword@mysite.redis.cache.windows.net:6380/0")],
                    },
        },
    }`

It would mean a lot, if anyone can provide with a source or docs that could be helpful

**Comment by carltongibson** [comment]
@vishwakarma0 Grrr. Sorry, no idea... — it's an Azure Redis connection issue. 

Do you need to use the `'rediss://...'` scheme for SSL maybe? 🤔

This isn't something we can help you with here I'm afraid. Good luck!

**Reply by KhushbooAgarwa** [reply]
I dont have words to thanks for this answer, I was trying to resolve this issue from months and just adding 's' in front of redis solved it

BIG THANKS

**Comment by vishwakarma0** [comment, marked as answer]
Hey @carltongibson 
Thanks a lot, it worked like a charm, all it needed was rediss in the url instead of redis.

**Reply by carltongibson** [reply]
Well there you are. 💃

**Reply by dreichen** [reply]
Hi @vishwakarma0 
I do not really know how to contact you. But it seems like I have a similar problem. I want to use django channels layer for a WebSocket connection. Seems like I do everything like you did (after the redis -> rediss think) but somehow I walways get "WebSocket rejected" after "Websocket handshake"... Do you have any example or method to contact you?
Best regards
